src/main/java/se/knowit/bookitnotification/service/NotificationServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package se.knowit.bookitnotification.service;
+
+import org.springframework.mail.MailException;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.scheduling.annotation.Async;
+import se.knowit.bookitnotification.model.NotificationValidator;
+import se.knowit.bookitnotification.model.Notification;
+
+public class NotificationServiceImpl implements NotificationService {
+
+    private JavaMailSender mailSender;
+
+    private NotificationValidator validator;
+
+    public NotificationServiceImpl(JavaMailSender mailSender) {
+        this.validator = new NotificationValidator();
+        this.mailSender = mailSender;
+    }
+
+    @Override
+    @Async
+    public void sendRegistrationNotificationAsync(Notification notification) {
+        try {
+            sendRegistrationNotification(notification);
+        } catch(IllegalArgumentException | MailException e) {
+            //TODO: log exception.
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void sendRegistrationNotification(Notification notification) throws IllegalArgumentException, MailException {
```

Comment:
Neither IllegalArgumentException or MailException are checked exception and therefore they don't need to be declared here.